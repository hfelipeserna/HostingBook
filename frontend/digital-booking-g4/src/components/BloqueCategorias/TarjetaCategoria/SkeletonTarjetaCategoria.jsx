import styles from "./TarjetaCategoria.module.css";

export default function SkeletonTajetaCategoria(){
    return (
        <div
            className={`${styles.tarjeta} ${styles.disabled}`}
        >
            <div
                className={`${styles.fotoPortada} ${styles.skeleton}`}
            ></div>
            <div className={styles.contenidoTarjeta}>
                <h2 className={`${styles.textLoader} ${styles.skeleton}`}></h2>
                <p className={`${styles.textLoader} ${styles.skeleton}`}></p>
            </div>
        </div>
    );
}