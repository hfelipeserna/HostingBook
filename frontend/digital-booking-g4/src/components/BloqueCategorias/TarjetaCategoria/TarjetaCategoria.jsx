import styles from "./TarjetaCategoria.module.css";

export default function TarjetaCategoria({
  fotoPortada,
  indice,
  estaActiva,
  nombre,
  descripcion,
  onToggleSelect,
}) {
  return (
    <div
      className={estaActiva ? styles.tarjetaSeleccionada : styles.tarjeta}
      onClick={() => onToggleSelect(indice, nombre)}
    >
      <div
        className={styles.fotoPortada}
        style={{
          backgroundImage: `url(${fotoPortada})`,
        }}
      ></div>
      <div className={styles.contenidoTarjeta}>
        <h2 className={styles.nombreCategoria}>{nombre}</h2>
        <p className={styles.descripcion}>{descripcion}</p>
      </div>
    </div>
  );
}
