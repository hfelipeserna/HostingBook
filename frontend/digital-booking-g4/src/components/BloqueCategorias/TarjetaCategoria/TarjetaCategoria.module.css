/*****************Tarjetas categoria*****************/
/*______Tarjeta entera______*/
.tarjeta,
.tarjetaSeleccionada {
  background-color: var(--color-5);
  background-position: top;
  border: 1px solid var(--color-4);
  border-radius: 10px;
  box-shadow: 0 4px 4px rgb(0 0 0 / 25%);
  height: 15.4rem;
  position: relative;
  transition: transform 0.5s;
  width: 20vw;
  min-width: 14rem;
  transition: all ease-in-out 300ms;
}
/*______Efecto tarjeta______*/
.tarjeta:hover,
.tarjetaSeleccionada {
  box-shadow: 4px 8px 12px rgb(0 0 0 / 25%);
  transform: scale(1.01);
  cursor: pointer;
}

.tarjetaSeleccionada {
  border: 2px solid var(--color-principal);
}

/*______Fondo tarjeta (portada)______*/
.fotoPortada {
  border-radius: 8px 8px 0 0;
  height: 75%;
  left: 0;
  position: absolute;
  top: 0;
  width: 100%;
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
}
/*______Contenedor/escrito______*/
.contenidoTarjeta {
  color: var(--color-2);
  position: relative;
  top: 79%;
}
/*______Nombre categoria______*/
.contenidoTarjeta h2 {
  /*.Hotel, .Hostel, .Departamento, .BedAndBreakfast */
  font-size: 1.25rem;
  font-style: normal;
  font-weight: 700;
  line-height: 1.438rem;
  margin-left: 3.87%;
  text-transform: capitalize;
  margin-bottom: 0.25rem;
}
/*______Descripcion categoria______*/
.contenidoTarjeta p {
  /*.descripcion*/
  font-size: 0.875rem;
  font-style: normal;
  font-weight: 700;
  line-height: 1rem;
  margin-left: 3.87%;
  opacity: 0.6;
}

/*****************Responsive*****************/
/*______Tablet______*/
@media screen and (max-width: 1045px) {
  .tarjeta,
  .tarjetaSeleccionada {
    width: 43vw;
    min-width: 12rem;
  }
}

@media screen and (max-width: 828px) {
  .tarjeta,
  .tarjetaSeleccionada {
    width: 42vw;
  }
}

/*______Mobile______*/
@media screen and (max-width: 480px) {
  .fotoPortada {
    height: 30vh;
  }
  .tarjeta,
  .tarjetaSeleccionada {
    height: 38vh;
    width: 90vw;
  }
  .contenidoTarjeta {
    top: 81%;
  }
}
