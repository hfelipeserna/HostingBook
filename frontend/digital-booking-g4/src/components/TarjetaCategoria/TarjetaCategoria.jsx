import styles from "./TarjetaCategoria.module.css";

export default function TarjetaCategoria(props) {
  return (
    <div className={styles.tarjeta}>
      {/* <img
        className={styles.fotoPortada}
        src={props.fotoPortada}
        alt="fondo de portada"
      ></img> */}
      <div
        className={styles.fotoPortada}
        style={{
          backgroundImage: `url(${props.fotoPortada})`,
        }}
      ></div>
      <div className={styles.contenidoTarjeta}>
        <h2 className={styles.hotel}>{props.nombre}</h2>
        <p className={styles.descripcion}>{props.descripcion}</p>
      </div>
    </div>
  );
}