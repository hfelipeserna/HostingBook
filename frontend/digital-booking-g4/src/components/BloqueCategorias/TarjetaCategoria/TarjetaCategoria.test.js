import React from "react";
import TarjetaCategoria from "./TarjetaCategoria";
import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/react"
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { createSerializer } from "enzyme-to-json";
import { configure, mount } from "enzyme";

configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));


describe("Tarjeta Categoria test", function () {
    const fotoPortada = "este es el url de la foto portada"
    let indice = 1
    let estaActiva = true
    const nombre = "este es el nombre"
    const descripcion = "esta es la descripcion"
    const onToggleSelect = (indice, nombre) => {
        if (estaActiva) {
            estaActiva = false
        } else {
            estaActiva = true
        }
    }

    test("Renderiza Por default", () => {
        const renderizacionSimple = render(<TarjetaCategoria></TarjetaCategoria>)
        expect(renderizacionSimple).toMatchSnapshot()
    })
    test("Renderiza Con data", () => {
        const renderizacionSimple = render(<TarjetaCategoria indice={indice} estaActiva={estaActiva} nombre={nombre} descripcion={descripcion} onToggleSelect={onToggleSelect}></TarjetaCategoria>)
        expect(renderizacionSimple).toMatchSnapshot()
    })
    test("Toggle on click", () => {
        const renderizacionSimple = mount(<TarjetaCategoria indice={indice} estaActiva={estaActiva} nombre={nombre} descripcion={descripcion} onToggleSelect={onToggleSelect}></TarjetaCategoria>)
        const botonToggle = renderizacionSimple.find(".tarjetaSeleccionada")

        botonToggle.simulate('click');
        renderizacionSimple.update()
        expect(estaActiva).toBeFalsy()
    })
})