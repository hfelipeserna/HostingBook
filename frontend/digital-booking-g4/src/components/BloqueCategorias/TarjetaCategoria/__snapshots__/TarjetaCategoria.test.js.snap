// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tarjeta Categoria test Renderiza Con data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="tarjetaSeleccionada"
      >
        <div
          class="fotoPortada"
          style="background-image: url(undefined);"
        />
        <div
          class="contenidoTarjeta"
        >
          <h2
            class="nombreCategoria"
          >
            este es el nombre
          </h2>
          <p
            class="descripcion"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="tarjetaSeleccionada"
    >
      <div
        class="fotoPortada"
        style="background-image: url(undefined);"
      />
      <div
        class="contenidoTarjeta"
      >
        <h2
          class="nombreCategoria"
        >
          este es el nombre
        </h2>
        <p
          class="descripcion"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Tarjeta Categoria test Renderiza Por default 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="tarjeta"
      >
        <div
          class="fotoPortada"
          style="background-image: url(undefined);"
        />
        <div
          class="contenidoTarjeta"
        >
          <h2
            class="nombreCategoria"
          />
          <p
            class="descripcion"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="tarjeta"
    >
      <div
        class="fotoPortada"
        style="background-image: url(undefined);"
      />
      <div
        class="contenidoTarjeta"
      >
        <h2
          class="nombreCategoria"
        />
        <p
          class="descripcion"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
